### ISSUE TITLE
using arrow keys after inserting a link breaks selection

### ISSUE THREAD

**Issue by arekkas** [ianstormtaylor/slate#363]
![broken links](http://g.recordit.co/W5kpYwbEeO.gif)

**Comment by arekkas** [comment]
@ianstormtaylor I was able to find what is causing this. After the link is inserted, the arrow keys seem to fail to trigger the `onSelect` callback in *components/content.js*. Any idea what this could be caused by? I would love to create a PR

**Comment by arekkas** [comment]
Ok, apparently the `onChange` callback in *components/editor.js* is not being called either when using the arrow keys after the link was inserted.

**Comment by arekkas** [comment]
Ok, apparently the state we get from one of the plugin responsible for handling onKeyDown is empty in onevent:

```js
  onEvent = (name, ...args) => {
    console.log('on event', name, args)
    for (const plugin of this.state.plugins) {
      if (!plugin[name]) {
        continue
      }
      const newState = plugin[name](...args, this.state.state, this)
      if (!newState) {
        // empty state
        continue
      }
      this.onChange(newState)
      break
    }
  }
```

**Comment by arekkas** [comment]
I think I found the issue. In `onKey[down|left|right|up]`, the `startText` is messed up:

```js
  function onKeyDownRight(e, data, state) {
    if (data.isCtrl) return
    if (data.isOpt) return
    if (state.isExpanded) return

    const { document, startText } = state
    const hasVoidParent = document.hasVoidParent(startText)

    if (
      startText.text == '' ||
      hasVoidParent
    ) {
      const nextText = document.getNextText(startText)
      if (!nextText) return state

      debug('onKeyDownRight', { data })

      e.preventDefault()
      return state
        .transform()
        .collapseToStartOf(nextText)
        .apply()
    }
  }
```

It seems like `startText.text` equals to the link title (e.g. "this is a link") instead of the paragraph's text, after inserting a link.

**Comment by arekkas** [comment]
Looks like this is caused by the state transform. Maybe `collapseToEnd`?

```
      state = state
        .transform()
        .wrapInline({
          type: 'link',
          data: { href }
        })
        .collapseToEnd()
        .apply()
```

**Comment by arekkas** [comment]
Ok, it seems to be a problem with `wrapInline`. Unfortunately this is where I hang, as I don't really know what's going on in there. I hope this trace helps you figure out the issue more quickly. :)

**Comment by arekkas** [comment]
This was resolved with 0.15
